models/points.go

```diff
@@ -167,7 +167,10 @@ func ParsePointsWithPrecision(buf []byte, defaultTime time.Time, precision strin
 		pos++
 
 		if len(block) == 0 {
-			break
+			if pos >= len(buf) {
```

Comment:
I think I was able to modify this to reduce the number of checks for the position to one. Your recommendation wouldn't work because it would have caused the last line to not be read, but I think this update should work properly.